src/document.js

```diff
@@ -1163,7 +1168,30 @@ function validateSetOptions(options) {
   return true;
 }
 
+/*!
+ * Verifies that 'obj' is a plain JavaScript object that does not have a custom
+ * encoding format. Objects that pass this check are encoded as 'Maps' in
+ * Firestore.
+ *
+ * @param {*} The argument to verify.
+ * @returns {boolean} 'true' if the input can be a treated as a plain object.
+ */
+function isPlainObject(obj) {
+  return (
+    is.object(obj) &&
+    !is.array(obj) &&
+    !is.date(obj) &&
+    !is.instance(obj, Firestore) &&
```

Comment:
If you keep this list of exclusions, I don't think Firestore belongs since this isn't something you can write in a document...  I'm also not sure if ResourcePath is valid?

Reply:
I am supporting ResourcePath. Firestore was added as a pre-caution.

But this is no longer relevant as per above.